### ISSUE TITLE
Editor opens two tabs "--crash-reporter-id" and "<the actual ID>" all the time

### ISSUE THREAD

**Issue by AndiDog** [microsoft/vscode#108806]
Issue Type: <b>Bug</b>

Since a few weeks back, this constantly happens. Screenshot linked from another person's complaint on [Reddit](https://www.reddit.com/r/vscode/comments/j1wwrg/what_is_crashreporterid_how_to_disable_it/):

![screenshot](https://preview.redd.it/4grmwevxg2q51.png?width=710&format=png&auto=webp&s=3d49c342f9838396866ab0d7389cb53a1a3cc208)

I disabled "Telemetry: Enable Crash Reporter" and it is still the case.

VS Code version: Code 1.50.1 (d2e414d9e4239a252d1ab117bd7067f125afd80a, 2020-10-13T14:53:05.704Z)
OS version: Darwin x64 19.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4558U CPU @ 2.80GHz (4 x 2800)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|10, 11, 10|
|Memory (System)|16.00GB (0.37GB free)|
|Process Argv|--crash-reporter-id 31c223cd-3acc-425e-bf4f-c87a9d3722dd|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (34)</summary>

Extension|Author (truncated)|Version
---|---|---
calculate|aca|2.1.0
slice|aik|0.0.3
transformer|dak|1.9.0
vscode-markdownlint|Dav|0.37.1
vscode-eslint|dba|2.1.10
xml|Dot|2.5.1
prettier-vscode|esb|5.7.1
hadolint|exi|0.3.0
go|gol|0.17.2
terraform|has|2.2.3
c-cpp-flylint|jbe|1.6.0
restructuredtext|lex|129.0.0
cppcheck|mat|0.2.2
vscode-kubernetes-tools|ms-|1.2.1
python|ms-|2020.9.114305
cpptools|ms-|1.0.1
vscode-typescript-tslint-plugin|ms-|1.2.3
uuid-generator|net|0.0.4
material-icon-theme|PKi|4.3.0
vscode-protolint|ple|0.5.1
vscode-yaml|red|0.11.1
rust|rus|0.7.8
vscode-nginx|sha|0.6.0
shellcheck|tim|0.12.0
reflow-paragraph|Tro|1.3.0
opa|tsa|0.6.1
cmake|twx|0.0.17
errorlens|use|3.2.3
vscode-apple-swift-format|vkn|1.1.1
vscode-todo-highlight|way|1.0.4
jinja|who|0.0.8
change-case|wma|1.0.0
clang-format|xav|1.9.0
vscode-proto3|zxh|0.5.2

(4 theme extensions excluded)

</details>

**Comment by deepak1556** [comment]
Can you provide steps to how you are opening code, the exact arguments used. Thanks!

**Comment by AndiDog** [comment]
Nothing super special. The app gets started normally from the macOS dock, or sometimes also through a keyboard shortcut configured in [Hammerspoon](https://github.com/Hammerspoon/hammerspoon) which does something very simple in code:

```sh
+(BOOL)launchByName:(NSString *)name {
    return [[NSWorkspace sharedWorkspace] launchApplication:name];
}
```

Here's the main process's tree currently:

```text
$ pstree 34707
-+= 34707 asommer /Applications/Visual Studio Code.app/Contents/MacOS/Electron
 |--- 34710 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (GPU).app/Contents/MacOS/Code Helper (GPU) --type=gpu-process --field-trial-handle=1718379636,7396363186260185226,16
 |--- 34712 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --type=utility --field-trial-handle=1718379636,7396363186260185226,165966158237181250
 |--- 34991 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --type=renderer --disable-color-correct-rendering --field-trial
 |-+- 43828 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --type=renderer --disable-color-correct-rendering --field-trial
 | |-+- 43835 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --inspect-port=0 /Applications/Visual Studio Code.app/Content
 | | |--- 01240 asommer /Users/asommer/.vscode/extensions/ms-python.python-2020.9.114305/languageServer.0.5.59/Microsoft.Python.LanguageServer
 | | |--- 43974 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app
 | | |--- 44003 asommer /Users/asommer/dev/extern/flutter/bin/cache/dart-sdk/bin/dart /Users/asommer/dev/extern/flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot --client-id=VS-Code --
 | | |-+- 44004 asommer bash /Users/asommer/dev/extern/flutter/bin/flutter daemon
 | | | \-+- 44019 asommer /Users/asommer/dev/extern/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev --packages=/Users/asommer/dev/extern/flutter/packages/flutter_tools/.packages /Users/asommer/dev/e
 | | |   \-+- 44331 asommer /usr/bin/script -t 0 /dev/null xcrun xcdevice observe --both
 | | |     \--= 44332 asommer /Applications/Xcode.app/Contents/Developer/usr/bin/xcdevice observe --both
 | | |--- 44042 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app
 | | |--- 44061 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Users/asommer/.vscode/extensions/dbaeumer.vscode-eslint-2.
 | | \--- 78984 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Users/asommer/.vscode/extensions/redhat.vscode-yaml-0.11.1
 | |--- 43837 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app/o
 | \--= 02656 asommer /usr/local/bin/bash -l
 \-+- 89430 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --type=renderer --disable-color-correct-rendering --field-trial
   |--- 36876 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app/o
   |-+- 89432 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --inspect-port=0 /Applications/Visual Studio Code.app/Content
   | |--- 11808 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Users/asommer/.vscode/extensions/dbaeumer.vscode-eslint-2.
   | |--- 11866 asommer /Users/asommer/.vscode/extensions/ms-python.python-2020.9.114305/languageServer.0.5.59/Microsoft.Python.LanguageServer
   | |--- 14526 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Users/asommer/.vscode/extensions/redhat.vscode-yaml-0.11.1
   | |--- 90322 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90323 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90324 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90325 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90326 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90327 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90328 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90329 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90330 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90331 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90332 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90334 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90335 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90336 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90337 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90338 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90339 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90340 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90341 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90342 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90343 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90344 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90345 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90346 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90347 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90348 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90349 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90350 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90351 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90352 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90353 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90354 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90355 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90356 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90357 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90358 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90359 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90360 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90361 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90362 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90363 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90364 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90365 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90366 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90367 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90368 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90369 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90370 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90371 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90372 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90373 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90374 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90375 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90376 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90377 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90378 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90380 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90381 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90384 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90385 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90386 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90387 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90388 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90389 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90390 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90391 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90392 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90393 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90394 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90395 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90396 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90397 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90398 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90399 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90400 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90401 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90402 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90403 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90404 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90405 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90406 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90407 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90408 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | |--- 90409 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   | \--- 90410 asommer /Users/asommer/.vscode/extensions/hashicorp.terraform-2.2.3/lsp/terraform-ls serve
   \--- 89437 asommer /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) /Applications/Visual Studio Code.app/Contents/Resources/app/o
```

**Comment by deepak1556** [comment]
Can you provide a screen recording, close those unsaved files and from a clean state try to open the app from the dock ?

We only append the `--crash-reporter-id` and the uuid value inside the app forwarding to a different process, it shouldn't be used for app startup.

This is only possible if whatever application is launching the app uses the following style of calls

```
open -b com.microsoft.VSCodeInsiders --args \ --crash-reporter-id abc-idsd
```

Can you also get the command line arguments used by `Hammerspoon` when launching the app ?

**Comment by AndiDog** [comment]
It's almost impossible to reliably reproduce. No command line arguments are used with the ways I start VS Code ([iTerm](https://github.com/gnachman/iTerm2/blob/286820c0965e466a10ac08a624d58a929dd8ab94/sources/iTermSemanticHistoryController.m#L223), Hammerspoon code from my above comment also doesn't look suspicious). And it sometimes happens rarely, sometimes more often. Is there a good place to debug it, such as specific logs?

**Comment by AndiDog** [comment]
Think I found the issue: I have one more app where VS Code is configured to open files (Git UI called "Fork"). I've configured it to run "code -- THE_FILE_PATH".

It's **reproducible** from the terminal: `code myfile.sql` works fine, but `code -- myfile.sql` (where `--` should protect from accidentally passing parameters instead of file paths) opens the file but then also opens the mentioned crash reporter related tabs.
